cnf-certification-test/lifecycle/suite.go

```diff
@@ -436,3 +441,24 @@ func testPodRequestsAndLimits(env *provider.TestEnvironment) {
 		ginkgo.Fail(errMsg)
 	}
 }
+
+func testPodPersistentVolumeReclaimPolicy(env *provider.TestEnvironment) {
+	ginkgo.By("Testing PersistentVolumes for reclaim policy to be set to delete")
+	var persistentVolumesBadReclaim []string
+
+	// Look through all of the pods, matching their persistent volumes to the list of overall cluster PVs and checking their reclaim status.
+	for _, put := range env.Pods {
+		for index := range env.PersistentVolumes {
+			if put.Data.Name == env.PersistentVolumes[index].Name && env.PersistentVolumes[index].Spec.PersistentVolumeReclaimPolicy != corev1.PersistentVolumeReclaimDelete {
```

Comment:
I don't understand this part of the check: `put.Data.Name == env.PersistentVolumes[index].Name`. If I understood PVs correctly, they are separate resources/objects with their own names, so this `if` probably won't match for any PV in the cluster, unless the customer is forced to use the same name for pods and PVs. Am I missing something?

Reply:
Ah yes!  You are correct.  I had to add a loop to look at the `Volumes` and their names.